main.tf

```diff
@@ -18,10 +18,24 @@ resource "azurerm_mssql_server" "vaconfigured" {
   version                      = "12.0"
   administrator_login          = "4dm1n157r470r"
   administrator_login_password = "4-v3ry-53cr37-p455w0rd"
-  minimum_tls_version          = "1.1"
+  minimum_tls_version          = "1.2"
 
   transparent_data_encryption_key_vault_key_id = azurerm_key_vault_key.vaconfigured.id
 
+
+  extended_auditing_policy {
+    storage_endpoint                        = "https://accurics.com"
+    retention_in_days                       = 120
+    storage_account_access_key              = "some-key"
+    storage_account_access_key_is_secondary = true
+  }
+
+  extended_auditing_policy {
+    storage_account_access_key              = "some-key"
+    storage_account_access_key_is_secondary = true
+    storage_endpoint                        = "https://accurics.com"
+    retention_in_days                       = 120
+  }
 }
```

Comment:
**Tenable.cs remediation**
 
**[MEDIUM]: Ensure key size is set on all keys for Azure Key Vault Key**
 
**Vulnerability**
Azure Key Vault Key does not have key size set on keys. This ensure key security is maintained.
 
**Remediation**
Key sizes cannot be altered once they are created. To create a new key with a more secure key size, follow the steps below.

In Azure Console -
1. Open the Azure Portal and go to Key Vaults.
2. Choose the Key vault you wish to edit.
3. Under Objects, select Keys.
4. Generate or upload a new key with the appropriate key size.

In Terraform - 
1. In the azurerm_key_vault_key resource, create keys with a key_size of 2048 or greater.

References:
[https://learn.microsoft.com/en-us/azure/key-vault/general/overview](https://learn.microsoft.com/en-us/azure/key-vault/general/overview)
[https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_key#key_size](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault_key#key_size)

source: tenablecs-